ethclient/ethclient.go

```diff
@@ -193,6 +194,25 @@ func (ec *Client) TransactionByHash(ctx context.Context, hash common.Hash) (tx *
 	return json.tx, json.BlockNumber == nil, nil
 }
 
+// BlockNumberByTransactionHash returns block number for the transaction with the given hash.
+func (ec *Client) BlockNumberByTransactionHash(ctx context.Context, hash common.Hash) (blockNumber uint64, err error) {
```

Comment:
Please call this method `TransactionInclusionBlock`.

Reply:
Thanks for reviewing my PR.
I renamed the method to `TransactionInclusionBlock`.
I see how I can return the block hash, using `json.BlockHash` or `json.txExtraInfo.BlockHash`, but I don't find where the transaction index is stored.
And last question, if I return the block hash and transaction index, how to get the blockNumber then?

Reply:
@fjl Should we also add this to some interface? If we want this functionality available in all our subsystems (e.g. abigen), we might need to make it a bit more extensive.

I can also add it later to ensure it plays nice with abigen, that's also fine (especially since abigen is very heavily modified in a pending pr).